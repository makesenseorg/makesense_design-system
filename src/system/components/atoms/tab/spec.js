import { shallowMount } from "@vue/test-utils";
import Comp from "./Tab.vue";

describe("Tab.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        id: "test-id",
        title: "test-id",
        active: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("sets class with active modifier set as true", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        id: "test-id",
        title: "test-id",
        active: true,
      },
    });
    expect(wrapper.classes()).toContain("tab-button--active");
  });

  it("emit id on click", async () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        id: "test-id",
        title: "test-id",
        active: true,
      },
    });

    wrapper.trigger("click");
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted().click).toBeTruthy();
  });
});
