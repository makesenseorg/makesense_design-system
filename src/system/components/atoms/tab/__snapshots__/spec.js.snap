// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab.vue matches snapshot 1`] = `
<div
  class="tab-button tab-button--active"
>
  
  test-id

</div>
`;
